### PR TITLE
Support alternate VHOST

### PR DESCRIPTION
**Pull request by dpetzel**
This is to address #12. It will allow overriding the default vhost, username and password via environment variables.

### CODE REVIEW

**Comment by yosifkit** [comment]
I have just a few suggestions:
- I would rather not even set any of these if they are blank, since then we'll just get the default from rabbit anyhow and not be specifying our own
- the file creation should probably be inside the `if` so that it only happens when running `rabbitmq-server`
  - there is a mixture of tabs and spaces in the heredoc, since it will be in the `if` block I would use `<<-` and indent with a tabs to match the block and then spaces to get nice formatting for the file ([heredoc](https://www.gnu.org/software/bash/manual/bashref.html#Here-Documents))

**Comment by dpetzel** [comment]
> I would rather not even set any of these if they are blank, since then we'll just get the default from rabbit anyhow and not be specifying our own. 

The problem I had with this was things got a little nasty. Since we are rendering the actual configuration file, not having a value means we shouldn't spit out that entire line. So if someone provide a username but not a password (or vice versa), there are now checks required for each line and we would need to conditionally append to the string. Its doable, it just lead to some nasty looking code. The defaults that are assigned are the defaults Rabbit would set itself (albeit they could in theory change those in the future)

> the file creation should probably be inside the if so that it only happens when running rabbitmq-server

That makes alot of sense to me, and your idea on the heredoc sounds promising as well. I don't do much herdocing in Bash, but that link you provided was helpful and looks easy enough.

**Comment by tianon** [comment]
I'm also +1 to creating the entire file conditionally -- having any of the vars present forcing the file to be created is fine IMO, but right now https://github.com/docker-library/rabbitmq/blob/14d58b0b10dce0a349cc2445bd4045da049df52a/Dockerfile#L41 makes `rabbitmq.config` something that users can overwrite in a depedent image or bind mount to enhance their configuration.  This change would break that entirely.  I'm not sure it's worth trying to "modify" an existing configuration to add these values, so IMO it's fine if we force folks to choose either supplying a custom config or using these environment variables.

File: docker-entrypoint.sh [hunk 1]
```diff
@@ -1,6 +1,22 @@
 #!/bin/bash
 set -e
 
+DEFAULT_VHOST=${RABBITMQ_DEFAULT_VHOST:-"/"}
+DEFAULT_USER=${RABBITMQ_DEFAULT_USER:-"guest"}
+DEFAULT_PASSWORD=${RABBITMQ_DEFAULT_PASSWORD:-"guest"}
```
Comment (tianon) [inline docker-entrypoint.sh]: These could be simplified with something like:
```bash
: ${RABBITMQ_DEFAULT_VHOST:=/}
: ${RABBITMQ_DEFAULT_USER:=guest}
: ${RABBITMQ_DEFAULT_PASSWORD:=guest}
```

Comment (dpetzel) [inline docker-entrypoint.sh]: That's awesome, I'm not much of a bash guy, and you just taught me something new. I like your approach way better.

**Pull request closed without merging**